docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,221 @@
+This command streamlines the `ilab data generate` pipeline and eliminates the requirement to define a `qna` document,
+which typically includes:
+* A minimum of 5×3 question-and-answer pairs.
+* Reference documents stored in Git.
+
+The goal is not to generate training data for InstructLab-trained models but to utilize the documents for RAG 
+workflows with pre-tuned models.
+
+#### Supported Databases
+The command may support various vector database types. A default configuration will align with the selected 
+InstructLab technology stack.
+
+#### Usage
+The generated embeddings can later be retrieved to enrich the context for RAG-based chat pipelines.
+
+#### Defining Command Options
+To maintain compatibility and simplicity, no new configurations will be introduced for this command. Instead,
+the settings will be defined using the following hierarchy (options higher in the list overriding those below):
+* CLI flags (e.g., `--rag`).
+* Environment variables following a consistent naming convention, such as `ILAB_<UPPERCASE_ARGUMENT_NAME>`.
+* Default values, for all the applicable use cases.
+
+For example, the `splitter_split_by` argument can be implemented using the `click` module like this:
+```py
+@click.option(
+    "--splitter-split-by",
+    default='word',
+    envvar="ILAB_SPLITTER_SPLIT_BY",
+)
+```
+
+### 2.3 Embedding Ingestion Pipeline Options
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Whether to include a transformation step. | `False` | `--transform` (boolean) | `ILAB_TRANSFORM` |
+| The output path of transformed documents (serve as input for the embedding ingestion pipeline). Mandatory when `--transform` is used. |  | `--transform-output` | `ILAB_TRANSFORM_OUTPUT` |
+| How to split the documents. One of `page`, `passage`, `sentence`, `word`, `line` | `word` | `--splitter-split-by` | `ILAB_SPLITTER_SPLIT_BY` |
+| Maximum number of units in each split. | `200` | `--splitter-split-length` | `ILAB_SPLITTER_SPLIT_LENGTH` |
+| Number of overlapping units for each split. | `0` | `--splitter-split-overlap` | `ILAB_SPLITTER_SPLIT_OVERLAP` |
+| Minimum number of units per split. | `0` | `--splitter-split-threshold` | `ILAB_SPLITTER_SPLIT_THRESHOLD` |
+| Vector DB implementation, one of: `milvuslite`, **TBD** | `milvuslite` | `--vectordb-type` | `ILAB_VECTORDB_TYPE` |
+| Vector DB service URI. | `./rag-output.db` | `--vectordb-uri` | `ILAB_VECTORDB_URI` |
+| Vector DB connection token. | | `--vectordb-token` | `ILAB_VECTORDB_TOKEN` |
+| Vector DB connection username. | | `--vectordb-username` | `ILAB_VECTORDB_USERNAME` |
+| Vector DB connection password. | | `--vectordb-password` | `ILAB_VECTORDB_PASSWORD` |
+| Name of the embedding model. | `sentence-transformers/all-minilm-l6-v2` | `--model` | `ILAB_EMBEDDING_MODEL_NAME` |
+| Token to download private models. |  | `--model-token` | `ILAB_EMBEDDING_MODEL_TOKEN` |
```

Comment:
This would introduce model downloading logic in a new place while `ilab model download` [already exists](https://github.com/instructlab/instructlab?tab=readme-ov-file#-download-the-model).

Reply:
Ah, good point!  I hadn't thought of that.  Using the existing model download sounds better to me.

Reply:
Sounds good to me. 
You mean that the user should first `ilab model download -rp sentence-transformers/all-minilm-l6-v2` and the RAG pipelines (both) would validate that a local download exists for that model before proceeding? 
Or would the RAG pipelines use the ilab download function to download the model locally?

Reply:
Perhaps the both most convenient and flexible solution would be to package in a default model but also allow downloading and configuring the use of a different model by doing `ilab model download...` and configuring it to be used.

Reply:
So it's a user responsibility to download it first 👍 
I'll add a note and also introduce a --model-dir option to supply a configurable location for looking for downloaded models